src/Platform.Handlers/src/Xamarin.Platform.Handlers/Hosting/AppBuilder.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Hosting.Internal;
+using Xamarin.Platform.Hosting.Internal;
+
+namespace Xamarin.Platform.Hosting
+{
+	public class AppBuilder : IAppHostBuilder
+	{
+		List<Action<HostBuilderContext, IHandlerServiceCollection>> _configureHandlersActions = new List<Action<HostBuilderContext, IHandlerServiceCollection>>();
+		List<Action<IConfigurationBuilder>> _configureHostConfigActions = new List<Action<IConfigurationBuilder>>();
+		List<Action<HostBuilderContext, IConfigurationBuilder>> _configureAppConfigActions = new List<Action<HostBuilderContext, IConfigurationBuilder>>();
+		List<Action<HostBuilderContext, IServiceCollection>> _configureServicesActions = new List<Action<HostBuilderContext, IServiceCollection>>();
+		List<IConfigureContainerAdapter> _configureContainerActions = new List<IConfigureContainerAdapter>();
+		IServiceFactoryAdapter _serviceProviderFactory = new ServiceFactoryAdapter<IServiceCollection>(new DefaultServiceProviderFactory());
+		bool _hostBuilt;
+		IConfiguration? _hostConfiguration;
+		IConfiguration? _appConfiguration;
+		HostBuilderContext? _hostBuilderContext;
+		HostingEnvironment? _hostingEnvironment;
+		IServiceProvider? _appServices;
+		IHandlerServiceProvider? _handlersServiceProvider;
+
+		public IDictionary<object, object> Properties => new Dictionary<object, object>();
+
+		public IHostBuilder RegisterHandlers(Dictionary<Type, Type> handlers)
+		{
+			foreach (var handler in handlers)
+			{
+				ConfigureHandlers((context, handlersCollection) => handlersCollection.AddTransient(handler.Key, handler.Value));
+			}
+
+			return this;
+		}
+
+		public IHostBuilder RegisterHandler<TType, TTypeRender>()
+			where TType : IFrameworkElement
+			where TTypeRender : IViewHandler
+		{
+			ConfigureHandlers((context, handlersCollection) => handlersCollection.AddTransient(typeof(TType), typeof(TTypeRender)));
+		
+			return this;
+		}
+
+		public (IHost Host, TApplication App) Init<TApplication>() where TApplication : class, IApp
```

Comment:
I wonder if there needs to be a separation between a `Host` and an `Application`. Why both - what's the difference?